File: CodingSamples/Java/Generics/TypeParamTest2/Program1.java
class Program {

	public static void main(String[] args) {
		SimpleStack<String> a = new SimpleStack<String>();
		a.push("Monday");
		a.push("Tuesday");
		a.push("Wednesday");
		a.push("Thursday");
		a.push("Friday");
		//a.push(12.3);
		while(!a.empty())
			System.out.println(a.pop());
		System.out.println("-----------------");
		SimpleStack<Double> b = new SimpleStack<>(); //type inference from declaration
		b.push(23.1);
		b.push(45.2);
		b.push(56.3);
		b.push(31.4);
		while(!b.empty())
			System.out.println(b.pop());
		System.out.println("-----------------");
		SimpleStack<Object> c = new SimpleStack<>();
		c.push("Sunday");
		c.push(97.5);
		c.push(1234);
		while(!c.empty())
			System.out.println(c.pop());
	}
}

File: CodingSamples/DotNet/Generics/TypeParamTest2/DemoApp/Program1.cs
class Program
{
    public static void Main(string[] args)
    {
        SimpleStack<string> a = new SimpleStack<string>();
        a.Push("Monday");
        a.Push("Tuesday");
        a.Push("Wednesday");
        a.Push("Thursday");
        a.Push("Friday");
        //a.Push(1.23);
        while(a.Count > 0)
            Console.WriteLine(a.Pop());
        Console.WriteLine("-----------------------");
        SimpleStack<Interval> b = new SimpleStack<Interval>();
        b.Push(new Interval(5, 31));
        b.Push(new Interval(7, 42));
        b.Push(new Interval(6, 53));
        b.Push(new (3, 14));
        while(b.Count > 0)
            Console.WriteLine(b.Pop());
        Console.WriteLine("-----------------------");
        SimpleStack<object> c = new SimpleStack<object>();
        c.Push("Sunday");
        c.Push(1.23);
        c.Push(new Interval(4, 25));
        while(c.Count > 0)
            Console.WriteLine(c.Pop());       
    }
}
